yspryn/hw9/fraction.py

```diff
@@ -0,0 +1,132 @@
+from math import gcd
+
+
+class Fraction:
+
+    def __init__(self, num: int, denom: int = 1):
+
+        if denom == 0:
+            raise ZeroDivisionError
+
+        _gcd = gcd(num, denom)
+        self.num = num // _gcd
+        self.denom = denom // _gcd
+
+        if self.denom < 0:
+            self.num *= -1
+            self.denom *= -1
+
+    @classmethod
+    def convert(cls, item):
+        if isinstance(item, int):
+            return cls(item, 1)
+        elif isinstance(item, cls):
+            return item
+
+    def __repr__(self):
+        return f"Fraction({self.num}, {self.denom})"
+
+    def __str__(self):
+        return f"\'{self.num} / {self.denom}\'"
+
+    # division
+    def __truediv__(self, other):
+        value1 = Fraction.convert(self)
```

Comment:
if you are not able to convert other, you should return 

NotImplemented    object as a result of operation. e.g.



Fraction(1, 2) / "abc"    # raises TypeError

assert Fraction(1, 2).__truediv__("abc") is NotImplemented